### ISSUE TITLE
Repository.createCommitOnHead ignores `filesToAdd` argument

### ISSUE THREAD

**Issue by freder** [nodegit/nodegit#655]
hi,

it looks like `Repository.createCommitOnHead` ignores the first argument `filesToAdd`. it is not using it at all, and instead [adds all the files!](https://github.com/nodegit/nodegit/blob/master/lib/repository.js#L496)

and that is why this function to create empty commits doesn't work as expected when there are (unstaged) files with changes:

``` javascript
function createEmptyCommit(repo, message, sig) {
    var files = []; // <--
    sig = sig || createSignature();
    var author = sig;
    var committer = sig;

    return unstageAll(repo) // make sure commit is in fact empty. (does a mixed reset)
        .then(function() {
            return repo.createCommitOnHead(
                files,
                author,
                committer,
                message
            );
        })
        .catch(function(err) {
            handleError(err);
        });
}
```

**Comment by freder** [comment]
using this for now

```javascript
function createCommitOnHead(filesToAdd, author, committer, message) {
	var index, repo;
	return getRepo()
		.then(function(repo_) {
			repo = repo_;
			return repo.openIndex();
		})
		.then(function(index_) {
			index = index_;
			index.read(1);
			if (!filesToAdd) { filesToAdd = []; }
			var result = filesToAdd.map(function(item) {
				return index.addByPath(item);
			});
			return Q.all(result);
		})
		.then(function() {
			index.write();
			return index.writeTree();
		}).then(function(treeOid) {
			return repo.getHeadCommit()
				.then(function(parent) {
					return repo.createCommit(
						'HEAD',
						author,
						committer,
						message,
						treeOid,
						[parent]
					);
				});
		}).
		catch(handleError);
};
```

**Comment by tbranyen** [comment]
Can you PR that?

**Comment by rcjsuen** [comment]
Can this be closed? #656 was closed but the corresponding #670 has been merged into `master`.
